### ISSUE TITLE
Do we actually need the `trimRange` method? 🤔

### ISSUE THREAD

**Issue by oleksandr-danylchenko** [recogito/text-annotator-js#66]
## TL;DR;
I think that we don't need to trim the selection range because:
1. User can intentionally _select non-textual nodes_. Taking them out of the selection range _only_ when they are at the edges, but not in the middle, seems like a partial prevention of a behavior
2. Ranges that have the `start/endContainer.nodeType` different from the `Node.TEXT_NODE` still can contain only the textual annotatable content. And such ranges are usually created when using the `.setStartAfter`/`. setEndBefore`

## Issue
In the https://github.com/recogito/text-annotator-js/issues/65 I brought up an issue with the `trimRange` method which is not that universal and cannot handle the trimming of the ranges that were split on `.not-annotatable` elements.
Then, in  https://github.com/recogito/text-annotator-js/issues/23 I noticed that `trimRange` doesn't act adequately as soon as the selection ends on a non-textual element: https://github.com/recogito/text-annotator-js/issues/23#issuecomment-1966301525.

---

All that led to the next **_conceptual questions about the purpose of the `trimRange` and whether we need it 🤔_**.

> The purpose of the method is to handle edge cases where the selection starts/ends on all sorts of non-text (images, layout block element, etc.) nodes. ©https://github.com/recogito/text-annotator-js/issues/23#issuecomment-1916179300

So the trim will discard the non-text nodes _only_ if they are _at the edges_ of the selection... but not in the middle... Although in both cases user _selected those nodes_.

For example, if I select the `text` + `img`, the latter will get removed. But if the selection is `text` + `img` + `text` - the image is kept in! Then why was it discarded in the first place 🤷🏻‍♂️. The same can be said for any other non-textual nodes that are trimmed on the edges, but not excluded from the middle of the selection. 
I feel like the not annotatable concept and the "selection with holes" fit such elements more 🤔 

---

Following that logic, I'm not sure about the robustness of the next checks we have in the code:
https://github.com/recogito/text-annotator-js/blob/3a70b9ef5cafec4b54ed629e934e5875d6f06262/packages/text-annotator/src/utils/trimRange.ts#L4-L9
https://github.com/recogito/text-annotator-js/blob/3a70b9ef5cafec4b54ed629e934e5875d6f06262/packages/text-annotator/src/state/spatialTree.ts#L42-L49
Why are we only interested in whether the edges of the selection are textual 🤔? Such a condition doesn't seem to represent that the range contains only the textual content... And, it eventually leads to an excessive `reviveSelector` runs - #64

---

Meanwhile, I don't think we should consider the ranges that have the `start/endContainer.nodeType` different from the `Node.TEXT_NODE` "invalid" or "non-textual". Using the convenient `.setStartAfter`/`.setEndBefore` browser usually sets the `start`/`end` to the `preceding`/`following` non-textual elements. Which still will be an accurate depiction of the textual range: (one of the ranges created by the `splitAnnotatableRanges`)
![image](https://github.com/recogito/text-annotator-js/assets/68850090/a33c0770-d0c3-4dc9-a447-16f3ac0c621b)
